### PR TITLE
support reduce op with fast implementation

### LINKED ISSUE #175: Fast reduce op reference support in benchgc

### PR DESCRIPTION
**Pull request by WangJialei-A**
Try to resolve #175

### CODE REVIEW

File: test/benchgc/src/benchgc/util.py [hunk 1]
```diff
@@ -131,16 +132,18 @@ def get_dtype(dtype: str) -> torch.dtype:
         return torch.bfloat16
     elif dtype == "u8" or dtype == "ui8":
         return torch.uint8
-    elif dtype == "s8" or dtype == "i8":
+    elif dtype in ["s8", "i8", "si8"]:
```
Comment (ZhennanQin) [inline test/benchgc/src/benchgc/util.py:135]: What does `si8` and `si32` mean?

Comment (WangJialei-A) [reply test/benchgc/src/benchgc/util.py:135]: signless -> i32/i8 
signed -> si32/si8
unsigned -> ui32/ui8

File: test/benchgc/src/benchgc/mlir/module.py [hunk 2]
```diff
@@ -46,3 +47,43 @@ def init_module(
                 block_args = f.entry_block.arguments
                 func.ReturnOp(op_func(ctx, *block_args))
         return module
+
+
+def init_reduce_module(
```
Comment (xurui1995) [inline test/benchgc/src/benchgc/mlir/module.py:52]: shall we add an `entry_name` param here? like `init_module`

**Review by xurui1995 — APPROVED** [review]
LGTM

**Review by ZhennanQin — APPROVED** [review]

**Pull request merged**